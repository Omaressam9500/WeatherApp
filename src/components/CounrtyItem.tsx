import React, { FC } from 'react';
import {
    TouchableOpacity,
    StyleSheet,
    Text,

} from 'react-native';
import { Colors, Pixel } from '../constants/styleConstatnts';
import { ICountry } from '../hooks/interfaces';



interface ICountryItem {
    countryItem: ICountry,
    onCountrySelect: (country: ICountry) => void,
    isLastIndex:boolean
}
export const CountryItem: FC<ICountryItem> = ({ countryItem,
    isLastIndex,
    onCountrySelect
}) => {
    const { name, region } = countryItem
    return (
        <TouchableOpacity style={[styles.mainContainer,isLastIndex?{borderBottomWidth:0}:null]}
            onPress={() => onCountrySelect(countryItem)}>
            <Text>{name} {","} {region}</Text>

        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    mainContainer: {
        paddingHorizontal: Pixel(20),
        paddingVertical: Pixel(10),
        borderBottomColor:Colors.disabledColor,
        borderBottomWidth:Pixel(1)

    }
})