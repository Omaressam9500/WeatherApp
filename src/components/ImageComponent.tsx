

import React, { FC, useContext, useEffect, useState, useRef } from "react";
import {
    StyleSheet, View,
} from "react-native";
import FastImage from 'react-native-fast-image'


interface IImage {
    url: string,
    style?: any;
    onLoadStart?: () => void,
    onLoadEnd?: () => void,
    onError?: () => void,
    defaultSource?: any,
    resizeMode?: any,
    isError?:boolean

}

export const ImageComponent: FC<IImage> = ({
    url,
    style,
    onLoadStart,
    onError,
    onLoadEnd,
    defaultSource,
    resizeMode,
    isError
}) => {
 
    return (<View>
        <FastImage
            style={[styles.imageStyle, style]}
       
            source={!isError ? {
                uri: url ,
                priority: FastImage.priority.normal,
                    cache: FastImage.cacheControl.immutable
            }
                : defaultSource}
            onLoadStart={onLoadStart}
            onLoadEnd={onLoadEnd}
            onError={onError}

            resizeMode={resizeMode ? resizeMode : FastImage.resizeMode.center}
        />
    </View>)
}
const styles = StyleSheet.create({
    imageStyle: { width: 50, height: 50, borderRadius: 25, marginBottom: 10 }
})
