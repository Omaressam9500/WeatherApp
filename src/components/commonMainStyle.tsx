import { I18nManager, StyleSheet } from "react-native";



const { isRTL } = I18nManager;


export const commonMainStyles = StyleSheet.create({

    rowBox: {
        flexDirection: "row",
        alignItems: "center",
    },
   

});


export const dir = isRTL ? "left" : "right";
