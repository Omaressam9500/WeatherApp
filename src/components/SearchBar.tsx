import React, { FC, useRef } from 'react';
import {

    StyleSheet,
    TextInput,
    FlatList,
    View,
} from 'react-native';
import { Colors, Pixel } from '../constants/styleConstatnts';
import { ICountry } from '../hooks/interfaces';
import { CountryItem } from './CounrtyItem';


interface ISearchHeader {
    handleSearchValue: (searchValue: string) => void;
    placeholder?: string,
    countriesList: Array<ICountry>
    onCountrySelect: (countryItem: ICountry) => void
}
const SearchBar: FC<ISearchHeader> = ({ handleSearchValue, countriesList, onCountrySelect
}) => {


    const textInputRef = useRef(null);

    const handleForceBlur = () => {
        textInputRef.current.blur();
    };
    const selectContryPress = (countryItem: ICountry) => {
        onCountrySelect(countryItem)
        handleForceBlur()
    }
    const renderItem = ({ item, index }) => (
        <CountryItem countryItem={item}
            isLastIndex={index === countriesList?.length - 1}
            onCountrySelect={selectContryPress
            } />
    );
    const onSearchResultClick = (text: string) => {
        handleSearchValue(text)
    }


    return (
        <View>
            <View style={[styles.container]}>
                <View style={{ flexDirection: 'row', justifyContent: 'flex-end', flex: 1 }}>
                    {
                        <View style={styles.textInputStyle}>
                            <TextInput style={styles.textInput}
                                ref={textInputRef}
                                testID={"input"}
                                placeholder={"Search Country"}
                                placeholderTextColor={"#919C9C"}
                                returnKeyType="search"
                                onChangeText={(text) => {
                                    onSearchResultClick(text)
                                }}
                                focusable={true}
                                onFocus={() => console.log('test')}
                            />
                        </View>
                    }

                </View>

            </View>
            {
                  countriesList?.length > 0 &&textInputRef.current.isFocused()&&
                <FlatList
                    data={countriesList}
                    renderItem={renderItem}
                    scrollEnabled={false}
                    keyExtractor={(item) => item?.id}
                    contentContainerStyle={styles.listStyle}
                />
            }

        </View>
    );
};


const styles = StyleSheet.create({
    container: {

        flexDirection: 'row',
    },
    searchSection: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },

    imagesView: {
        flexDirection: 'row',
        alignItems: 'flex-end'
    },
    textInputStyle: {
        backgroundColor: Colors.sacandAppBackgroundColor,
        borderRadius: Pixel(55),
        borderColor: '#001F5F',
        borderWidth: 1,
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        paddingRight: Pixel(10),
        paddingLeft: Pixel(15),
        height: Pixel(48),
        marginRight: Pixel(7.5),

        fontSize: Pixel(14),
    },
    textInput: {
        flex: 1,
        color: "#202020",
        fontSize: Pixel(14)

    },

    icnView: {
        height: Pixel(48),
        width: Pixel(48),
        borderRadius: Pixel(24),
        alignItems: 'center',
        justifyContent: 'center'
    },
    icons: {
        width: Pixel(24),
        height: Pixel(24),
        backgroundColor: '#FBFBFB',
        borderRadius: Pixel(5),
        alignItems: 'center',
        justifyContent: 'center',
    },
    listStyle: {
        backgroundColor: Colors.sacandAppBackgroundColor,
        borderRadius: Pixel(15),
        marginTop: Pixel(10),
        borderTopColor:Colors.sacandAppBackgroundColor,
        borderTopLeftRadius:0,
        borderTopRightRadius:0,
        marginHorizontal:Pixel(5),

        borderColor: Colors.disabledColor,
        borderWidth: Pixel(1)
    },
});

export default SearchBar;
