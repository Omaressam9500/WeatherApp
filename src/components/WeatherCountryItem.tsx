

import React, { FC } from 'react';
import {
    View,
    StyleSheet,
    Text,

} from 'react-native';
import { Colors, Pixel } from '../constants/styleConstatnts';
import { ICountry, IHour } from '../hooks/interfaces';
import { commonMainStyles } from './commonMainStyle';
import { ImageComponent } from './ImageComponent';



interface IWeatherCountryItem {
    conditionItem: IHour,

}
export const WeatherCountryItem: FC<IWeatherCountryItem> = ({ conditionItem,
  
}) => {
    const {condition,temp_f } = conditionItem
    return (

        <View style={commonMainStyles.rowBox}>
            <ImageComponent
                url={`https:${condition?.icon}`} />
            <Text style={style.conditionTxt}> {condition?.text}  {temp_f} {"Â°F"}</Text>

        </View>
    )
}

const style = StyleSheet.create({

    conditionTxt: {
        marginLeft: Pixel(10)
    }
})