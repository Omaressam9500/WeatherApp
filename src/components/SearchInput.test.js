import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SearchBar from './SearchBar';
import {WeatherCountryItem} from '../components/WeatherCountryItem';
import renderer from 'react-test-renderer';
test('Should apply the value when changing text', () => {
  const handleSearchValue = jest.fn()
  const { getByTestId } = render(<SearchBar handleSearchValue={handleSearchValue} countriesList={[]} onCountrySelect={() => { }} />);
  const input = getByTestId('input');
  fireEvent.changeText(input, "test");
  expect(input.props.value).toBe('test');
  expect(handleSearchValue).toHaveBeenCalledWith('test');
});

test('WeatherCountryItem snapshot test', () => {
const data = {
    condition: {
      text: 'Partly cloudy',
      icon: '//cdn.weatherapi.com/weather/64x64/day/116.png',
    },
    temp_f: 77.4,
  };


  const tree = renderer.create(<WeatherCountryItem conditionItem={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});

