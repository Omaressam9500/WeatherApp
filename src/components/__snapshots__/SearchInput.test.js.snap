// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherComponent snapshot test 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "marginLeft": 17,
      }
    }
  >
     
      
     
    Â°F
  </Text>
</View>
`;
